### PR TITLE
Add Fire Hydrant connector

### PR DESCRIPTION
**Pull request by huongtn**
## Description

> Provide description here

## Type of change
- [ ] Bug fix
- [ ] New feature
- [ ] Breaking change

## Related issues

> Fix [#1]() 

## Migration notes

> Describe migration notes if any

## Extra info

> Add any additional information

### CODE REVIEW

File: destinations/faros-destination/src/converters/firehydrant/incidents.ts [hunk 1]
```diff
@@ -0,0 +1,289 @@
...
+    const incident = record.record.data as Incident;
+    const res: DestinationRecord[] = [];
+
+    const maxDescriptionLength = this.maxDescriptionLength(ctx);
+
+    const applicationMapping = this.applicationMapping(ctx);
+    const incidentRef = {uid: incident.id, source};
+    const createdAt = Utils.toDate(incident.created_at);
+    const updatedAt = incident.events
+      ? Utils.toDate(incident.events[incident.events.length - 1].occurred_at)
```
Comment (willmarks) [inline destinations/faros-destination/src/converters/firehydrant/incidents.ts:53]: What if incident.events is empty?

File: destinations/faros-destination/src/converters/firehydrant/incidents.ts [hunk 2]
```diff
@@ -0,0 +1,289 @@
...
+        service.name in applicationMapping &&
+        applicationMapping[service.name].name
+      ) {
+        const mappedApp = applicationMapping[service.name];
+        application = {
+          name: mappedApp.name,
+          platform: mappedApp.platform ?? application.platform,
+        };
+      }
+      res.push({model: 'compute_Application', record: application});
```
Comment (willmarks) [inline destinations/faros-destination/src/converters/firehydrant/incidents.ts:144]: I don't think we want to write an application unless it is mapped. Lets move this and the `ims_IncidentApplicationImpact` into this if statement.

File: destinations/faros-destination/resources/spec.json [hunk 3]
```diff
@@ -439,6 +439,29 @@
...
+                    "properties": {
+                      "application_mapping": {
+                        "type": "string",
+                        "title": "Application Mapping",
+                        "description": "JSON map of FireHydrant service(s) name, to compute platform specific app name and platform name. Used to reference compute_Application object, from an ims_Incident object.",
+                        "multiline": true,
+                        "default": "{}",
+                        "examples": [
+                          "{ \"Aion\": { \"name\": \"aion\", \"platform\": \"ECS\" } }"
+                        ]
```
Comment (willmarks) [inline destinations/faros-destination/resources/spec.json:460]: do we need the `max_description_length` here?

File: destinations/faros-destination/src/converters/firehydrant/incidents.ts (hunk 1 above)
Comment (huongtn) [reply destinations/faros-destination/src/converters/firehydrant/incidents.ts:53]: yes, incident.event never be empty

File: destinations/faros-destination/src/converters/firehydrant/incidents.ts [hunk 4]
```diff
@@ -0,0 +1,289 @@
...
+      };
+
+      const occurredAt = Utils.toDate(event.occurred_at);
+      if (
+        !resolvedAt &&
+        event.data.current_milestone === FirehydrantIncidentMilestone.resolved
+      ) {
+        resolvedAt = occurredAt;
+        eventType.category = IncidentEventTypeCategory.Resolved;
+      }
```
[the remark this answers is not in the document]
Comment (huongtn) [reply destinations/faros-destination/src/converters/firehydrant/incidents.ts:72]: the newest event always at the top

File: destinations/faros-destination/resources/spec.json (hunk 3 above)
Comment (huongtn) [reply destinations/faros-destination/resources/spec.json:460]: ok, I added already

File: destinations/faros-destination/src/converters/firehydrant/incidents.ts (hunk 2 above)
Comment (huongtn) [reply destinations/faros-destination/src/converters/firehydrant/incidents.ts]: ok , thanks!

**Comment by huongtn** [comment]
Hi @willmarks, Can you help to advice for test issues:
Non-abstract class 'FirehydrantIncidents' does not implement inherited abstract member 'source' from class 'FirehydrantConverter'.
It just got issue when I updated yesternight, it worked before

**Comment by willmarks** [comment]
Hey @huongtn, I believe the issue is related to [this](https://github.com/faros-ai/airbyte-connectors/blob/main/destinations/faros-destination/src/converters/converter.ts#L12-L13). It looks like all converters need to supply the `source` rather than the `source` being inferred from the class name.

**Review by willmarks — APPROVED** [review]
LGTM

**Pull request merged**